module_utils/stcrest.py

```diff
@@ -59,12 +59,14 @@ def new_session(self, user_name, session_name, reset_existing=True, kill_existin
             params = {'userid': user_name, 'sessionname': session_name}
             rsp = conn.post(url, data=params, timeout=60 * 2)
             log.info("SESSION %s %s -> [%d] %s" % (url, json.dumps(params, indent=4), rsp.status_code, rsp.content))
-
             if rsp.status_code != 409 and rsp.status_code != 200 and rsp.status_code != 201:
                 log.error("Failed to create a session: %d %s" % (rsp.status_code, rsp))
                 return False
 
         self.session = sessionID
+
+        self.conn.headers.update({'Accept': 'application/json', "X-STC-API-Session": self.session})
```

Comment:
Is this really needed? The session is already set here:

https://github.com/Spirent/stc-ansible/blob/3d4046d436237a7281c447489e4d7259316013bf/module_utils/stcrest.py#L36

Reply:
When do reset action, the header will not be updated. It will use the previous session name.

Reply:
> Is this really needed? The session is already set here:
> 
> https://github.com/Spirent/stc-ansible/blob/3d4046d436237a7281c447489e4d7259316013bf/module_utils/stcrest.py#L36

When do reset action, the header will not be updated. It will use the previous session name.

Reply:
Ok. Can you then remove then `, "X-STC-API-Session": self.session` from line 36?

Reply:
> Ok. Can you then remove then `, "X-STC-API-Session": self.session` from line 36?

Ok. Let me test it after remove it.

Reply:
> Ok. Can you then remove then `, "X-STC-API-Session": self.session` from line 36?

Done.